File: javascript/leetcode/121. Best Time to Buy and Sell Stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  const sz = prices.length;
  if (!sz) return 0;

  let max = 0;
  let min = prices[0];

  for (let i = 1; i < sz; i++) {
    if (prices[i] < min) min = prices[i];
    max = max > prices[i] - min ? max : prices[i] - min;
  }
  return max;
};

File: c++/leetcode/121. Best Time to Buy and Sell Stock.cpp
class Solution
{
public:
  int maxProfit(vector<int> &prices)
  {
    const int sz = prices.size();
    if (!sz)
      return 0;

    int max = 0;
    int min = prices[0];

    for (int i = 1; i < sz; i++)
    {
      if (prices[i] < min)
        min = prices[i];
      max = max > prices[i] - min ? max : prices[i] - min;
    }

    return max;
  }
};